pkg/controller/installation/validation.go

```diff
@@ -54,6 +56,26 @@ func validateCustomResource(instance *operatorv1.Installation) error {
 				),
 			)
 		}
+
+		// validate logging config for calico-cni
+		if instance.Spec.Logging != nil && instance.Spec.Logging.CNILogging != nil {
```

Comment:
I think we have one last semantic problem with this API we need to at least conceptualize a solution to (if not implement), which is how to let users turn off file logging altogether.

Right now, it's impossible for a user to set these field to `nil` (because our defaulting will fill them in) and it's impossible for a user to set these fields to zero (because our validation will error). 

So basically there's no way to turn off the logging altogether. 

It's a tricky situation - usually, we would say that specifying `spec.Logging.CNILoggin: nil` means "disable this feature", however we need to be backwards compatible and doing this would mean that old clusters would lose file logging on upgrade, so our defaulting needs to fill this in.

We might need a `fileLogging: Enabled` field in this struct that defaults to `Enabled` but can be set to `Disabled`.

That said, there's no way to disable file loggingt _today_ prior to your changes, so this isn't a regression and I think maybe we shouldn't block this PR on this. Just wanted to call it out so you're aware.

Reply:
That's a good idea - I looked at how cni-plugin is setting up the logging and looks like this is a feature that needs to be implemented on cni-plugin as well. Having that in mind, as you said, it makes more sense for this change to be done separately. Should I create a ticket to track this task?

I am thinking we need to define the fileLoggingEnabled flag on both cni-plugin and operator api. on way, is to not overwrite the values of other fields and just ignore them when fileLoggingEnabled flag is ON. the flag could be a shortcut to not setting / changing any loggingconfig values on operator side and not setting up a fileLogger on cni-plugin side.

What do you think?

Reply:
I'd raise a backlog item for it, yeah.

file logging enabled in the CNI plugin appears to be configured by the `log_file_path` option, according to my read of this code: https://github.com/projectcalico/calico/blob/1ffe1b75b4c2b24bdd308b9150ede44ed3b555a4/cni-plugin/internal/pkg/utils/utils.go#L739-L770

Reply:
ticket created here: https://tigera.atlassian.net/browse/EV-3421?atlOrigin=eyJpIjoiOWYwZWI3NmUyZTZiNDMyYWEyMjg1MjA0MjBjZmE2OWMiLCJwIjoiaiJ9